test/helper.lua

```diff
@@ -3,101 +3,219 @@ local fio = require("fio")
-                {type = "TREE", parts = { {7, type = "str", path = "data[*].name", is_nullable = true} }})
+        space:create_index("json_path_index", {
+            type = "TREE",
+            parts = {
+                {
+                    field = 6,
+                    type = "scalar",
+                    path = "age",
+                    is_nullable = true
+                }
+            }
+        })
+        space:create_index("multikey_index", {
+            type = "TREE",
+            parts = {
+                {
+                    field = 7,
+                    type = "str",
+                    path = "data[*].name"
+                }
+            }
+        })
+        if engine ~= "vinyl" then
+            space:create_index("functional_index", {
+                type = "TREE",
+                parts = {
+                    {
+                        field = 1,
+                        type = "string"
+                    }
+                },
+                func = "tree_func"
+            })
+        end
     end
 
-    local bitset = box.schema.create_space("bitset", { if_not_exists = true })
-    bitset:create_index("primary", {type = "TREE", parts={ 1 }})
-    bitset:create_index("index_for_first_name",
-            {type = "BITSET", parts={ {field = 2, type = "string"} }, unique = false})
-end)
+    return space
+end
+
+function helpers.create_space_with_hash_index(engine)
+    local space = create_space("hash", engine)
+    space:create_index("primary", {
+        type = "HASH",
+        parts = {
+            {
+                field = 1
+            }
+        }
+    })
+    space:create_index("index_for_first_name", {
+        type = "HASH",
+        parts = {
+            {
+                field = 2
+            }
+        }
+    })
+    space:create_index("multipart_index", {
+        type = "HASH",
+        parts = {
+            {
+                field = 1
+            },
+            {
+                field = 2
+            }
+         }
+     })
+
+    return space
+end
+
+function helpers.create_space_with_bitset_index(engine)
+    local space = create_space("bitset", engine)
+    space:create_index("primary", {
+        type = "TREE",
+        parts = {
+            {
+                field = 1
+            }
+        }
+    })
+    space:create_index("index_for_first_name", {
+        type = "BITSET",
+        parts = {
+            {
+                field = 2,
+                type = "string"
+            }
+        },
+        unique = false
+    })
+
+    return space
+end
 
 t.after_suite(function()
     fio.rmtree(t.datadir)
 end)
 
-function helpers.init_spaces(g)
-    g.tree = box.space.tree
-    g.hash = box.space.hash
-    g.vinyl = box.space.vinyl
-    g.bitset = box.space.bitset
-end
+t.before_suite(function()
+    t.datadir = fio.tempdir()
+    box.cfg{
+        wal_dir    = t.datadir,
+        memtx_dir  = t.datadir,
+        vinyl_dir  = t.datadir,
+        vinyl_memory = 1024,
```

Comment:
I guess this value requires a comment. Seems it is needed to get yields during select.

Reply:
I don't know real reasons to change `vinyl_memory` here. @ArtDu could you elaborate?

Reply:
This is needed to test `test_mvcc_vinyl_tx_conflict`. It can probably be tested differently. Oleg suggested this option, I did not come up with an alternative. 
https://github.com/tarantool/expirationd/blob/fdff71bdfa6c92240420f2262a1c7241a9dc350a/test/unit/atomic_iteration_test.lua#L125
But I did not do it very well, in the sense that this value can only be increased at runtime, but not decreased, IIRC. Therefore, I only set the value back after running `test_kill_task`. And this is a crutch

https://github.com/tarantool/expirationd/blob/fdff71bdfa6c92240420f2262a1c7241a9dc350a/test/unit/atomic_iteration_test.lua#L161-L175

Reply:
Added a comment:

```
commit 57e12153ab34c805f04229553cd0eb8ccbac4fea
Author: Sergey Bronnikov <sergeyb@tarantool.org>
Date:   Thu Nov 11 16:29:25 2021 +0300

    Add explanation for vinyl_memory option
    
    Part of #80

diff --git a/test/helper.lua b/test/helper.lua
index ee2977d..3b8e25c 100644
--- a/test/helper.lua
+++ b/test/helper.lua
@@ -89,6 +89,7 @@ t.before_suite(function()
         wal_dir    = t.datadir,
         memtx_dir  = t.datadir,
         vinyl_dir  = t.datadir,
+        -- vinyl_memory is changed to test test_mvcc_vinyl_tx_conflict.
         vinyl_memory = 1024,
     }
 
```